modules/gplazma2-pyscript/src/main/java/org/dcache/gplazma/pyscript/PyscriptPlugin.java

```diff
@@ -42,6 +43,35 @@ private static List<Path> listDirectoryContents(String directoryPath) throws IOE
         return Files.list(Path.of(directoryPath)).toList();
     }
 
+    private void executePythonFile(Path path) throws IOException {
+        String content = Files.lines(path).collect(
+                Collectors.joining("\n")
+        );
+        // execute file content, loading the function "py_auth" into the namespace
+        PI.exec(content);
+    }
+
+    private static List<String> principalSetToStringList(Set<Principal> principals) {
+        if (principals.isEmpty()) {
+            // empty set handling
+            return new ArrayList<String>();
```

Comment:
better to use `Collections.emptyList()`

Reply:
because it returns an immutable empty list that is memory efficient?

Reply:
Right, it returns a singleton, thus, doesn't allocate new memory when an empty list is required.